components/apps.js

```diff
@@ -649,33 +647,151 @@ SteamUser.prototype.getOwnedDepots = function(excludeSharedLicenses) {
- * has been emitted.
- * @param {boolean} [excludeSharedLicenses=false] - Pass true to exclude licenses that we have through family sharing
+ * Returns an array of licenses this account owns.
  * @returns {int[]}
  */
-SteamUser.prototype.getOwnedPackages = function(excludeSharedLicenses) {
-	if (this.steamID.type != SteamID.Type.ANON_USER && !this.licenses) {
+SteamUser.prototype.getOwnedLicenses = function() {
+	if (!this.licenses) {
 		throw new Error("We don't have our license list yet.");
 	}
 
+	return this.licenses;
+};
+
+/**
+ * Returns an array of package IDs this account owns (different from owned licenses). The filter only
+ * works, if enablePicsCache option is enabled and appOwnershipCached event has been emitted.
+ * @param {object|function} filter - Options for what counts for ownership, or a custom filter function
+ * @returns {int[]}
+ */
+SteamUser.prototype.getOwnedPackages = function(filter) {
+	// We're anonymous
 	if (this.steamID.type == SteamID.Type.ANON_USER) {
 		return [17906];
 	}
 
 	// We're an individual user
-	let packages = this.licenses;
-	if (excludeSharedLicenses) {
-		packages = packages.filter(license => license.owner_id == this.steamID.accountid);
+	let packages = this.getOwnedLicenses();
+
+	// If passed filter argument is empty, and global ownership filter is set, use it
+	if (!filter && this.options.ownershipFilter) {
+		filter = this.options.ownershipFilter;
+	}
+
+	// Support for the old syntax and behavior (<= v4.19.12)
+	if (!filter) {
+		// Slight deviation of old behavior: don't return expired licenses
+		return this._returnPackages(packages);
+	}
+	// If filter options is a boolean, we asssume it's excludeSharedLicenses
+	if (typeof filter === 'boolean') {
+		let excludeSharedLicenses = filter;
+		filter = {excludeShared: excludeSharedLicenses};
+	}
+
+	// New behavior from this point on
+	// Handle filter options
+	if (typeof filter === 'object') {
+		const defaults = {
+			excludeFree: false, // By default, include free licenses (FreeOnDemand, NoCost, etc.)
+			excludeShared: false, // By default, include shared licenses
+			excludeExpiring: false // By default, include licenses that are going to expire in the future (free weekends)
+		};
+		filter = Object.assign(defaults, this.options.ownershipFilter, filter);
+
+		// If there is nothing to filter, no pics cache needed
+		if (!filter.excludeFree && !filter.exludeShared && !filter.excludeExpiring) {
+			return this._returnPackages(packages);
+		}
+
+		// We don't need pics cache for only excludeShared
+		if (!filter.excludeFree && filter.exludeShared && !filter.excludeExpiring) {
+			packages = packages.filter(license => license.owner_id == this.steamID.accountid);
+			return this._returnPackages(packages);
+		}
+	}
+
+	// From this point, we need pics cache to be enabled
+	this._ensurePicsCache();
+
+	// Determine the filter function
+	let packageFilter = null;
```

Comment:
`licenseFilter`? 🤔

Reply:
Technically true, but packages and licenses are used interchangeably here. I think `packageFilter` is fine, as the whole function is about handling and returning packages.